lib/libyui_client/widgets/base.rb

```diff
@@ -103,6 +104,11 @@ def collect_all
       def class_name
         self.class.name.split('::').last
       end
+
+      def find_widgets
+        LibyuiClient.logger.info("Search for #{class_name} #{@filter}")
+        @widget_controller.find(@filter.plain).body(regex_filter: @filter.regex)
```

Comment:
Are those regex executed every time?

So for this line I understand that we always send query to the server with the part of the filter which is not regex and always try filter the body of the response with the part of the filter which is a regex (if exists).
I can see that you updated the filter in `action` although I still don't get why we don't do it when reading property, perhaps this is related...

This regex processing in the client looks like it should run only the first time that any method on the widget is called. Once we resolve the regex in the filter the first time, the rest of the calls should access those values. I'm afraid the user will not try to introduce in the most normal case, plain and regex, just the regex for the id for example, when string to identify it is really weird, does it mean that it would request the whole widget json each time?

Would be extensible this design to be able to do that and save bandwidth/processing?

Reply:
> does it mean that it would request the whole /widgets json each time?

No. It happens only in case you don't have any plain filter.

Reply:
> > does it mean that it would request the whole /widgets json each time?
> 
> No. It happens only in case you don't have any plain filter.

exactly, which IMO would be common case when somebody use a regex, so each time we call a property we need to run it again, instead of doing only it the first time.

Reply:
discussed offline that we will evaluate needs in the future for that enhancement.